### PR TITLE
MTDSA-15290- GTS added for Historic non-FHL UK Property Business Annual Submission

### PR DESCRIPTION
**Pull request by rhysconnor**
MTDSA-15290- GTS added for Historic non-FHL UK Property Business Annual Submission endpoints

### CODE REVIEW

File: resources/public/api/conf/2.0/historic_uk_non_fhl_annual_submission_delete.yaml [hunk 1]
```diff
@@ -26,6 +26,10 @@ description: |
               <td><p>NOT_FOUND</p></td>
               <td><p>Simulates the scenario where no data is found.</p></td>
           </tr>
+          <tr>
+              <td><p>STATEFUL_DELETE</p></td>
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_non_fhl_annual_submission_delete.yaml:30]: oh this is new. Are we planning to call it STATEFUL_DELETE now?
Why are we not keeping it consistent and just use STATEFUL ?

Comment (rhysconnor) [reply resources/public/api/conf/2.0/historic_uk_non_fhl_annual_submission_delete.yaml:30]: The reason this is STATEFUL_DELETE is that the delete for this endpoint is a PUT, not a DELETE, so we needed to create a new GTS so we could allow for the logical delete

Comment (QHMRC) [reply resources/public/api/conf/2.0/historic_uk_non_fhl_annual_submission_delete.yaml:30]: After speaking to Rhys on the call, I understand this is used in order to to achieve some backend logic.

Though, I still believe we should keep it consistent. Using STATEFUL_DELETE and STATEFUL will be confusing on the vendors side

Comment (QHMRC) [reply resources/public/api/conf/2.0/historic_uk_non_fhl_annual_submission_delete.yaml:30]: I've created a ticket: https://jira.tools.tax.service.gov.uk/browse/MTDSA-15574

**Review by QHMRC — APPROVED** [review]

**Pull request closed without merging**